src/Model/CompletionMeet.rei

```diff
@@ -6,30 +6,26 @@ open CamomileBundled.Camomile;
 // (where the candidate completions _meet_ the text). It includes
 // the position where completions should be requested from, as well
 // as the 'base' - candidate text that should apply to the filtering of completion items.
-type t;
+type t = {
+  bufferId: int,
+  // Base is the prefix string
+  base: string,
+  // Meet is the location where we request completions
+  location: Location.t,
+};
```

Comment:
I don't see a good reason to hide this. The purpose of this data structure is pretty much just to bundle data together. Hiding it and adding constructor and accessor functions for everything is just extra boilerplate.

There are two motivation for doing encapsulation (at least that I can think of right now):

1. When you want to be able to change the data structure without breaking consumer code.
2. When you want to maintain an invariant internally, that is to restrict the possible values the data can gave.

Neither of these seem to apply in this case, and a few other cases I've seen, such as `Buffer`.

Reply:
Agree regarding `CompletionMeet`!

`Buffer`, though, the encapsulation makes sense I think - we'll likely pursue a different underlying implementation for the data structure (it is a naive array-of-strings right now, but that isn't the ideal data structure for performance) - long-term I picture us moving to a different structure, like a rope or piece table (IE, #80 ) - so having an abstraction like `getLine` means that we can change the underlying backing data store w/o needing the consumer to change the way they interact with `Buffer`s

Reply:
There's no need to hide the entire data structure just because the structure of one field is prone to change. We can hide the structure of just that field instead. But yeah, that's absolutely a good reason for encapsulation.

(And also really interesting data structures to explore!)